pallets/collator-assignment/src/lib.rs

```diff
@@ -670,6 +671,35 @@ pub mod pallet {
             }
         }
     }
+
+    impl<T: Config> PendingCollatorAssignmentHelper<T::AccountId> for Pallet<T> {
+        fn remove_offline_collator_from_pending_assignment(collator: &T::AccountId) {
+            // If there is a  pending assignment, we remove the collator from it
```

Comment:
@tmpolaczyk I would say we cannot do that. we cannot remove them from pending assignment nor current. that would disturb current assignments and therefore harm existing chains. what we have to do is simply mark it as non-online in pallet-staking, making it non-eligible anymore in the next sessions